src/v/model/model.cc

```diff
@@ -176,13 +176,17 @@ std::filesystem::path ntp::topic_path() const {
 std::istream& operator>>(std::istream& i, compression& c) {
     ss::sstring s;
     i >> s;
-    c = string_switch<compression>(s)
-          .match_all("none", "uncompressed", compression::none)
-          .match("gzip", compression::gzip)
-          .match("snappy", compression::snappy)
-          .match("lz4", compression::lz4)
-          .match("zstd", compression::zstd)
-          .match("producer", compression::producer);
+    try {
+        c = string_switch<compression>(s)
+              .match_all("none", "uncompressed", compression::none)
+              .match("gzip", compression::gzip)
+              .match("snappy", compression::snappy)
+              .match("lz4", compression::lz4)
+              .match("zstd", compression::zstd)
+              .match("producer", compression::producer);
+    } catch (std::runtime_error& e) {
```

Comment:
maybe could avoid the try/catch with

string_switch<std::optional<compression>>(c)
...
.default_match(std::nullopt);

if (!c.has_value()) {
  i.setstate(...)
}

Reply:
fair point. force of habit from regular switch, but static analysis is no help here obviously.